cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -319,6 +320,27 @@ public boolean checkAnyRunning(Predicate<Id.Program> predicate, ProgramType... t
   }
 
   @Override
+  public Set<Id.Program> getAnyRunning(Predicate<Id.Program> predicate, ProgramType... types) {
+    Set<Id.Program> runningPrograms = new HashSet<>();
+    for (ProgramType type : types) {
+      for (Map.Entry<RunId, ProgramRuntimeService.RuntimeInfo> entry :  list(type).entrySet()) {
+        ProgramController.State programState = entry.getValue().getController().getState();
+        if (programState.isDone()) {
+          continue;
+        }
+        Id.Program programId = entry.getValue().getProgramId();
```

Comment:
The `Id.Program` is deprecated. For any newly written class/method, use `ProgramId` instead.